lib/pagination/paginate.js

```diff
@@ -0,0 +1,29 @@
+const adapter = require('./adapter');
+const config = require('./config');
+const logger = require('../logger/index');
+
+const page = options => (options.page === undefined ? config.defaultPage : options.page);
+const limit = options => (options.limit === undefined ? config.defaultLimit : options.limit);
+
+const content_pagination = currentAdapter => ({
+  page: currentAdapter.paginated_content(),
+  count: currentAdapter.count(),
+  total_pages: currentAdapter.total_pages(),
+  total_count: currentAdapter.total_count(),
+  previous_page: currentAdapter.previous_page(),
+  current_page: currentAdapter.page,
+  next_page: currentAdapter.next_page(),
+  previous_page_url: currentAdapter.pageUrl(currentAdapter.previous_page()),
+  next_page_url: currentAdapter.pageUrl(currentAdapter.next_page())
+});
+
+exports.paginate = (content, req, options) => {
+  const currentAdapter = adapter;
+  try {
+    currentAdapter.createAdapter(content, page(options), limit(options), req);
```

Comment:
You create and adapter and after make a pagination, you can't simply paginate in each request directly and create adapter one single time ? I imagine use of pagination like this

```
const myPaginator = createPaginator(options);
....code
const somethingPaginated1 = myPaginator.paginate(something1);
const somethingPaginated2 = myPaginator.paginate(something2);
...more code
```

Reply:
The purpose of having an 'adapter' object is that the object represents the pagination request each time. If I try to paginate something with 25 results, there will be a different paginator than If I request just 5 results. Every paginator object has to be different, and created just for that request, so the paginator is unique to the parameters sent to the request, and behaves according to those parameters and to that request and not to another. If we just called the paginator once, then all the pagination requests would follow the same flow and would be very hard to let  the user change that flow and paginate however he wants.